pkg/cmd/cli/cmd/process.go

```diff
@@ -214,14 +212,17 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 		// when user specify the --value
 		if cmd.Flag("value").Changed {
 			values := kcmdutil.GetFlagStringSlice(cmd, "value")
-			injectUserVars(values, cmd, obj)
+			if err := injectUserVars(values, obj); err != nil {
+				return err
```

Comment:
> screwed up some parameter and was not able to re-run the command again as some of the objects had been created already.

I will point out that is exactly why you can label `oc process` and why `oc delete` takes a file input (`oc process | oc delete -f -` will tidy up).

If you stop this from accepting multiple templates, then the point of contention goes away.